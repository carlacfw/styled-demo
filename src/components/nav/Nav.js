import React, { Component } from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'
import colours from '../../shared/colour'

const NavWrapper = styled.div`
  display: flex;
  flex-direction: column;
  min-width: 10%;
  height: 100%;
  background-color: ${colours.primary};
  border-right: solid 2px ${colours.secondary};
`

const Ul = styled.ul`
  list-style: none;
  padding: 10px;
`

const Li = styled.li`
  margin: 10px 0;
  border: 1px solid grey;
  background-color: ${colours.secondary};
`

const StyledLink = styled(NavLink).attrs({
  activeClassName: 'activeClassName'
})`
  display: block;
  padding: 5px;
  color: #444444;
  background: #ccc;
  text-decoration: none;
  margin: 3px;
  &:hover {
    background-color: lightcyan;
  }
  &.selected {
    background-color: lightblue;
  }
`

const Button = styled.button`
  background-color: ${colours.secondary};
  width: 26px;
  height: 26px;
`
class Nav extends Component {
  constructor(props) {
    super(props)
    this.state = {
      open: true
    }
  }

  navDisplay() {
    if (this.state.open) {
      return (
        <NavWrapper>
          <Button
            onClick={() => {
              this.setState({ open: false })
            }}
          >
            X
          </Button>
          <Ul>
            <Li>
              <StyledLink to="/" exact activeClassName="selected">
                Home
              </StyledLink>
            </Li>
            <Li>
              <StyledLink to="/form" activeClassName="selected">
                Form
              </StyledLink>
            </Li>
            <Li>
              <StyledLink to="/props" activeClassName="selected">
                Props
              </StyledLink>
            </Li>
          </Ul>
        </NavWrapper>
      )
    } else {
      return (
        <Button
          onClick={() => {
            this.setState({ open: true })
          }}
        >
          ...
        </Button>
      )
    }
  }
  render() {
    return this.navDisplay()
  }
}

export default Nav
