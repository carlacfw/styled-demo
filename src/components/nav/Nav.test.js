import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Nav from './Nav'
describe('The Nav ', () => {
  it('It renders the Navu correctly', () => {
    const wrapper = shallow(<Nav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
