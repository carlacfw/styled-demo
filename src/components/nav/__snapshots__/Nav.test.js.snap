// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Nav  It renders the Navu correctly 1`] = `
<styled.div>
  <styled.button
    onClick={[Function]}
  >
    X
  </styled.button>
  <styled.ul>
    <styled.li>
      <Styled(NavLink)
        activeClassName="selected"
        exact={true}
        to="/"
      >
        Home
      </Styled(NavLink)>
    </styled.li>
    <styled.li>
      <Styled(NavLink)
        activeClassName="selected"
        to="/form"
      >
        Form
      </Styled(NavLink)>
    </styled.li>
    <styled.li>
      <Styled(NavLink)
        activeClassName="selected"
        to="/props"
      >
        Props
      </Styled(NavLink)>
    </styled.li>
  </styled.ul>
</styled.div>
`;
